### ISSUE TITLE
std::format create undefined-behavior [UndefinedBehaviorSanitizer] [label=c++20]

### ISSUE THREAD

**Issue by nathan-in-code** [llvm/llvm-project#118805]
```c++
#include <format>

int main()
{
  auto const str = std::format("{0}", -1);
}

``` 

If this mini-code is compiled with “AddressSanitizer” option, an undefined-behavior is detected.
The error only occurs with **negative Integers**.

Clang++-Version = 18
Option =  -fsanitize=unsigned-integer-overflow
OS: Ubuntu 24.04 (64-bit)

example: 
```shell
clang++-18 -fsanitize=unsigned-integer-overflow  -std=c++20 format.cpp -o my_format
```
output:
 runtime error: negation of 4294967295 cannot be represented in type 'make_unsigned_t<int>' (aka 'unsigned int')
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior /usr/bin/../lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/format:1099:12

**Comment by philnik777** [comment]
Works fine with libc++: https://godbolt.org/z/vn45K3n91

**Comment by nathan-in-code** [comment]
Interesting, when I run the code in godbolt.org,
I get the same error again. (see below right.)

![Image](https://github.com/user-attachments/assets/aa3b18ce-8803-4f04-bda4-4cd8cb86bab6)

**Comment by nathan-in-code** [comment]
Ok, then I'll have to install clang-19.

Because clang-18 (default on Ubuntu 24.04) has compiler issues with libc++ at std::formater.

**Comment by frederick-vs-ja** [comment]
The "bug" can be fixed if we change [this line](https://github.com/gcc-mirror/gcc/blob/5289540ed58e42ae66255e31f22afe4ca0a6e15e/libstdc%2B%2B-v3/include/std/format#L1174) to an equivlanent but arcane form.

```diff
-	    __u = -static_cast<make_unsigned_t<_Int>>(__i);
+	    __u = make_unsigned_t<_Int>{1} + ~static_cast<make_unsigned_t<_Int>>(__i);
```

I personally _think_ that -fsanitize=unsigned-integer-overflow is somehow insane. It's now working by design, but I'm not sure whether we should inform libstdc++ to get well with it.
